src/app.ts

```diff
@@ -277,6 +282,61 @@ app.use(async (ctx, next) => {
     await next();
 });
 
+app.use(async (ctx, next) => {
+    const request = ctx.req;
+    const host = request.header('host');
+    if (!host) {
+        // TODO handle
+        throw new Error('No Host header');
+    }
+    ctx.set('role', 'Anonymous');
+
+    const authorization = request.header('authorization');
+
+    if (!authorization) {
+        return next();
+    }
+
+    const [match, token] = authorization.match(/Bearer\s+(.*)$/) || [null];
+
+    if (!match) {
+        throw new Error('Invalid Authorization header');
+    }
+
+    let protocol = 'https';
+    // We allow insecure requests when not in production for things like testing
+    if (process.env.NODE_ENV !== 'production' && !request.raw.url.startsWith('https')) {
+        protocol = 'http';
+    }
+
+    const jwksURL = new URL('/ghost/.well-known/jwks.json', `${protocol}://${host}`);
+
+    const jwksResponse = await fetch(jwksURL, {
+        redirect: 'follow'
+    });
+
+    const jwks = await jwksResponse.json();
+
+    const key = await jose.JWK.asKey(jwks.keys[0]);
+
+    try {
+        const claims = jwt.verify(token, key.toPEM());
+        if (typeof claims === 'string' || typeof claims.role !== 'string') {
+            return;
```

Comment:
This will result in no role being set, we probably want it to be `Anonymous` like if an error was thrown right?

Reply:
Yessir - thanks